Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs

```diff
@@ -293,6 +308,87 @@ private void OnValidateCommand(ValidateCommandEvent evt)
             }
         }
 
+        private string GetPreviousActionNameFromViewTree(in ActionOrBindingData data)
+        {
+            Debug.Assert(data.isAction);
+
+            // If TreeView currently (before delete) has more than one Action, select the one immediately
+            // above or immediately below depending (if data is first in the list)
+            var treeView = ViewStateSelector.GetViewState(stateContainer.GetState()).treeViewData;
+            if (treeView.Count > 1)
+            {
+                string actionName = data.name;
+                int index = treeView.FindIndex(item => item.data.name == actionName);
+                if (index > 0)
+                    index--;
+                else
+                    index++;
+
+                return treeView[index].data.name;
+            }
+
+            return string.Empty;
```

Comment:
what happens if the action was not found/the treeview is empty and the string is empty? I think in this case nothing should get selected?

Reply:
If `actionName` isn't found then `FindIndex()` returns -1 which gets incremented to 0 in the `else` branch causing the first Action in the list to be selected. Typically I don't like taking this kind of dependency on an API and prefer to write more "defensive" code, but in this case `List<T>.FindIndex()` is a well established .NET API that won't change.

Still, I added a comment to clarify this reasoning with fbacf6f958456ecc949d89a22634f77574cfc607.

Reply:
And if the action list is empty? (Edge case, but still possible)

Reply:
The outer condition `if (treeView.Count > 1)` handles this case, i.e. if the Action list is empty this will return 0 and we jump to the default return.
But even if we didn't check, `FindIndex()` would still return -1 if the list was empty (didn't match actionName).